source/agora/consensus/Ledger.d

```diff
@@ -936,42 +936,42 @@ public class Ledger
 
     ***************************************************************************/
 
-    private string validateBlockSignature (in Block block) @safe nothrow
+    public string validateBlockSignature (in BlockHeader header) @safe nothrow
```

Comment:
This assumes `block.hashFull() == block.header.hashFull()`, should there be a test validating that?

Reply:
Same goes for `Nominator.signBlock()`

Reply:
I will add one.

Reply:
In fact there is one here:- https://github.com/bosagora/agora/blob/4269bf72d6b146a24d70a518bd8fa7006dcec9fa/source/agora/consensus/data/Block.d#L728